stac_generator/raster_data/generator.py

```diff
@@ -29,105 +39,133 @@ def __init__(
         catalog_cfg: StacCatalogConfig | None = None,
         href: str | None = None,
     ) -> None:
+        # Reset the DataFrame index
         source_df = source_df.reset_index(drop=True)
-        if 'epsg' in source_df.columns:
-            source_df['epsg'] = source_df['epsg'].astype(int)
+
+        # Ensure the EPSG column is properly formatted
+        if "epsg" in source_df.columns:
+            source_df["epsg"] = source_df["epsg"].astype(int)
+
+        # Parse the 'bands' column into a list if it is a JSON-encoded string
+        if "bands" in source_df.columns:
+
+            def parse_bands(bands: Any) -> Any:
+                if isinstance(bands, str):
+                    try:
+                        return json.loads(bands)
+                    except json.JSONDecodeError as e:
+                        raise ValueError(f"Error parsing 'bands': {bands}") from e
+                return bands
+
+            source_df["bands"] = source_df["bands"].apply(parse_bands)
+
+        # Pass the modified DataFrame to the base class
         super().__init__(
             source_df=source_df,
             collection_cfg=collection_cfg,
             catalog_cfg=catalog_cfg,
             href=href,
         )
 
-    def create_item_from_config(self, source_cfg: RasterSourceConfig) -> List[pystac.Item]:
-        logging.info(f"Processing raster from: {source_cfg.location}")
-        
+    def create_item_from_config(self, source_cfg: RasterSourceConfig) -> list[pystac.Item]:
+        logging.info("Processing raster from: %s", source_cfg.location)
+
         try:
             with rasterio.open(source_cfg.location) as src:
                 bounds = src.bounds
                 bbox = [bounds.left, bounds.bottom, bounds.right, bounds.top]
                 transform = src.transform
                 crs = src.crs
-                
-                geometry = {
-                    "type": "Polygon",
-                    "coordinates": [[
-                        [bounds.left, bounds.bottom],
-                        [bounds.left, bounds.top],
-                        [bounds.right, bounds.top],
-                        [bounds.right, bounds.bottom],
-                        [bounds.left, bounds.bottom]
-                    ]]
-                }
-
-                # Create base item
+                shape = [src.height, src.width]
```

Comment:
You should be able to get the shape from the geotiff above with `shape = src.shape`. I think this way is preferable because it removes the possibility of getting the ordering wrong here.

Reply:
done